codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/Version.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.smithy
+
+import software.amazon.smithy.codegen.core.CodegenException
+
+class Version(private val content: String) {
+    fun fullVersion(): String = content
+
+    fun crateVersion(): String {
+        val parts = content.split("-")
+        if (parts.size < 2) {
+            return ""
+        }
+        return parts.first()
```

Comment:
If you want to go for "the file contains one line", this may not always work; some versions don't follow the format `vx.y.z`, like `release-2022-08-04`. In this case you should remove the last element, instead of picking the first one